auth/access.js

```diff
@@ -0,0 +1,36 @@
+const config = require('./../config');
+const jwt = require('jsonwebtoken');
+const { crudService, errorService } = require('./../services');
+const { UserModel } = require('./../models');
+const { URLS } = require('./../shared/constants');
+
+const access = (req, res, next) => {
+    try {
+        const token = req.headers.authorization.split(' ')[1];
+
+        jwt.verify(token, config.jwt.privateKey, (err, payload) => {
+            if (payload) {
+                crudService
+                    .findById(UserModel, res, payload.userId, false)
+                    .then(doc => {
+                        req.user = doc;
+                        next();
+                    })
+            } else {
+                next();
+            }
+        });
+    } catch (error){
+        let access = false;
+
+        URLS.access.forEach(item => {
```

Comment:
you better split this middleware into two :
 1. check if user is authorized
 2. check permission

Reply:
OK. I'll split